.teamcity/src/main/kotlin/common/extensions.kt

```diff
@@ -114,7 +115,11 @@ fun BuildType.applyDefaultSettings(os: Os = Os.LINUX, arch: Arch = Arch.AMD64, b
     paramsForBuildToolBuild(buildJvm, os, arch)
     params {
         // The promotion job doesn't have a branch, so %teamcity.build.branch% doesn't work.
-        param("env.BUILD_BRANCH", "%teamcity.build.branch%")
+        param(
+            "env.BUILD_BRANCH",
+            if (this@applyDefaultSettings is CompileAll) "%teamcity.build.branch%"
+            else "%dep.Gradle_${DslContext.uuidPrefix}_Check_CompileAllBuild.teamcity.build.branch%"
```

Comment:
❌ I'd only do this for the functional tests that use the parallel tests feature. I am pretty sure there are some builds that use `applyDefaultSettings` but don't even depend on `CompileAll`.